ocean_lib/models/dtfactory.py

```diff
@@ -54,12 +56,12 @@ def get_token_address(self, transaction_id: str) -> str:
                 f"Cannot get the transaction receipt for tx {transaction_id}."
             )
             return ""
-
-        logs = getattr(self.events, "TokenRegistered")().processReceipt(tx_receipt)
+        logs = getattr(self.events, "TokenRegistered")().processReceipt(
```

Comment:
Since we're touching this code anyway, lets upgrade it to use the ContractEvents API instead of `getattr`.  This is better because it helps IDEs navigate the codebase quicker. Like so:

```python
logs = self.events.TokenRegistered().processReceipt(tx_receipt, errors=DISCARD)
```